splash/browser_tab.py

```diff
@@ -503,6 +505,29 @@ def runjs(self, js_source, handle_errors=True):
         js_source = "%s;undefined" % js_source
         self.evaljs(js_source, handle_errors=handle_errors)
 
+    def runjs_async(self, js_source, callback, errback, timeout=0):
+        """
+        An async variant of runjs.
+
+        It injects a callback function named <lua_resume> into the scope of
+        the user's script. When lua_resume() is called, the corresponding
+        Lua coroutine will resume.
```

Comment:
Isn't is an async variant of evaljs? Some other ideas for the function name: 

* `evaljs_async`
* `wait_for_value` 
* `wait_for_return`